consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/ExecutionPlanRepositoryImpl.scala

```diff
@@ -134,6 +134,41 @@ class ExecutionPlanRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends
           edges = edges)
     }
   }
+
+  override def getDataSources(value: String, access: Option[String])(implicit ec: ExecutionContext): Future[Array[String]] ={
+    val readResult = db.queryStream[String](
+      """
+        FOR d in dataSource
+        |FOR dp in depends
+        |FILTER dp._from == @planId AND dp._to == d._id
+        |RETURN d.uri
+        |""".stripMargin
+        .stripMargin,
+      Map("planId" -> "executionPlan".concat("/").concat(value)
+      )
+    )
+
+    val writeResult = db.queryStream[String](
+      """
+        FOR d in dataSource
+        |FOR aff in affects
+        |FILTER aff._from == @planId AND aff._to == d._id
+        |RETURN d.uri
+        |""".stripMargin
+        .stripMargin,
+      Map("planId" -> "executionPlan".concat("/").concat(value)
```

Comment:
This is not the best way for querying a graph in ArangoDB.
Your query involves two scans with filter (full collection scan + index scan + filter).
In this case a graph traversal would be way more efficient.
Also it's advisable to use document keys (without collection name prefix) instead of full IDs outside the database. In this query you know that you are looking at the `executionPlan` collection, so passing it as a parameter makes no sense.

Consider the following query:
```aql
FOR ds IN 1..1 
    OUTBOUND DOCUMENT('executionPlan', @planKey) affects
    RETURN ds.uri
```

Reply:
makes sense. I am making the change.